src/xdp/programinspect.c

```diff
@@ -1154,7 +1166,7 @@ XdpProgramValidateRule(
     //
     RtlZeroMemory(ValidatedRule, sizeof(*ValidatedRule));
 
-    if (UserRule->Match < XDP_MATCH_ALL || UserRule->Match > XDP_MATCH_TCP_CONTROL_DST) {
+    if (UserRule->Match < XDP_MATCH_ALL || UserRule->Match > XDP_MATCH_IP_NEXT_HEADER) {
```

Comment:
I dislike code like this that you much change for every new enum. It's easy to miss. Let's define `XDP_MATCH_COUNT` that has a comment indicating it must be last, and then fix this to use that.

Reply:
Jeff Tippet recommended us not to define _MAX (we did, before) because it makes it easy for other code not in sync with the latest headers to make false assumptions. With this model, the risk is entirely contained within XDP.

Reply:
I'd consider adding an internal-only header definition, but aside from this single line of code, such a macro _hides_ the fact that each point of maintenance is also specific to each enum value.

Reply:
I'll respectfully disagree with Jeff here. You have two places that must be manually identified by the dev every time a new one of these is added, with no compile-time way to catch. Ensuring the code is correct is more important than possibly giving a false assumption to an app IMO. That's my opinion here, and it's how all the QUIC code is written (where this kind of thing matters) and it's never been a problem, only help.

Reply:
This was a conversation with Jeff several years ago, but he had seen the pain in NDIS and other decades-old Windows headers with countless binaries interacting with it. Since QUIC is relatively new and obscure, it may simply not have changed enough headers to hit problems.

There is exactly one line in all of XDP that is safe to not look at as a developer when changing the enum.

Reply:
I won't die on this hill, but I'd still change it if it was me.